c7n/query.py

```diff
@@ -225,6 +229,28 @@ def augment(self, resources):
             results = list(w.map(_augment, chunks(resources, self.chunk_size)))
             return list(itertools.chain(*results))
 
+    @property
+    def account_id(self):
+        """ Return the current account ID. 
+        
+        This should now be passed in using the --account-id flag, but for a
+        period of time we will support the old behavior of inferring this from
+        IAM.
+        """
+        if self.config.account_id:
+            return self.config.account_id
+
+        # Below this comment is deprecated.
+        msg = ("Warning: Inferring the account ID from IAM is deprecated. "
+               "Use the --account-id flag to specify this value.")
+        self.log.warning(msg)
+        print(msg, file=sys.stderr)
```

Comment:
@kapilt - Looks like `DeprecationWarning` is ignored by default.  I enabled it using the line below.  Do you want this in place?  (Seems like weird behavior that it has to be explicitly enabled)

```
warnings.simplefilter('always', DeprecationWarning)
```

As for validation - I did add the check at validation time for the S3 attach-encrypt action.  But you are right that it is a runtime problem for the others.